### PR TITLE
Release v2.7.1

### PR DESCRIPTION
**Pull request by grst**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/scrnaseq/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/scrnaseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,14 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## v2.7.1 - 2024-08-12
```
Comment (jfy133) [inline CHANGELOG.md:6]:
```suggestion
## v2.7.1 - 2024-08-13
```

File: workflows/scrnaseq.nf [hunk 2]
```diff
@@ -67,7 +71,8 @@ workflow SCRNASEQ {
     ch_salmon_index = params.salmon_index ? file(params.salmon_index) : []
 
     //star params
-    ch_star_index = params.star_index ? file(params.star_index) : []
+    star_index = params.star_index ? file(params.star_index, checkIfExists: true) : null
+    ch_star_index = star_index ? [[id: star_index.baseName], star_index] : []
```
Comment (jfy133) [inline workflows/scrnaseq.nf:75]: Is it correct to call this a `ch_` prefix when it's not actually in a channel object yet (it seems)? 

I'm assuming you're fine with the code, I'm just checking something hasn't been missed

**Review by jfy133 — APPROVED** [review]
No blockers from me!

**Review by mashehu — APPROVED** [review]

**Comment by grst** [comment]
image
👀 

#358

File: workflows/scrnaseq.nf (hunk 2 above)
Comment (jfy133) [reply workflows/scrnaseq.nf:75]: No, no standards. 

Just checking that a channel generation wasn't missing anywhere, where it should have been :)

**Pull request merged**